tests/testSpanSets.py

```diff
@@ -292,7 +292,30 @@ def testIntersectNot(self):
 
         self.assertEqual(outputSpanSet, expectedSpanSet)
 
-
+        numIntersectNotTrials = 100
+        spanRow = 5
+        for N in range(numIntersectNotTrials):
+            # Create two random SpanSets, both with holes in them
+            listOfRandomSpanSets = []
+            for i in range(2):
+                # Make two rectangles to be turned into a SpanSet
+                rand1 = np.random.randint(0, 26, 2)
+                rand2 = np.random.randint(rand1.max(), 51, 2)
+                tempList = [afwGeom.Span(spanRow, rand1.min(), rand1.max()),
+                            afwGeom.Span(spanRow, rand2.min(), rand2.max())]
+                listOfRandomSpanSets.append(afwGeom.SpanSet(tempList))
+
+            # IntersectNot the SpanSets, randomly choosing which one is the one
+            # to be the negated SpanSet
+            randChoice = np.random.randint(0, 2)
+            negatedRandChoice = int(not randChoice)
+            sourceSpanSet = listOfRandomSpanSets[randChoice]
+            targetSpanSet = listOfRandomSpanSets[negatedRandChoice]
+            resultSpanSet = sourceSpanSet.intersectNot(targetSpanSet)
+            for span in resultSpanSet:
+                for point in span:
+                    self.assertTrue(sourceSpanSet.contains(point))
+                    self.assertFalse(targetSpanSet.contains(point))
```

Comment:
Almost, but not complete. This checks that every point in the result should be there. But it does not check that every point that should be part of the result is (for instance, this will always work if the result is empty, even when it shouldn't). Thus it is better to do the result check on the pixel level.